compute_endpoint/globus_compute_endpoint/engines/globus_compute.py

```diff
@@ -269,6 +269,7 @@ def start(
         # Add executor to poller *after* executor has started
         self.job_status_poller = JobStatusPoller(**self._job_status_kwargs)
         self.job_status_poller.add_executors([self.executor])
+        self._engine_ready = True
```

Comment:
Consider using one or more other components as a proxy for engine readiness.

E.g.,
```python
@property
def _ready(self):
    return hasattr(self, "job_status_poller")
```

Reply:
I originally hoped to check the status of `self.executor`, `self._status_report_thread` and `self.job_status_poller` as a proxy for engine readiness, but these objects do not have publicly accessible interfaces to do so. The quick example I wrote above is functionally the same as a simple flag, so all good here.